exonum/src/runtime/versioning.rs

```diff
@@ -157,32 +157,35 @@ use crate::runtime::{ArtifactId, CoreError, ExecutionError, ExecutionFail};
 /// // Requirements can be parsed from a string.
 /// let req: ArtifactReq = "some.Service@^1.3.0".parse().unwrap();
 ///
-/// let valid_artifact = ArtifactId {
-///     runtime_id: RuntimeIdentifier::Rust as _,
-///     name: "some.Service".to_owned(),
-///     version: "1.5.7".parse().unwrap(),
-/// };
+/// let valid_artifact = ArtifactId::new(
+///     RuntimeIdentifier::Rust as u32,
+///     "some.Service".to_owned(),
+///     "1.5.7".parse().unwrap(),
+/// ).unwrap();
 /// req.try_match(&valid_artifact).unwrap();
 ///
 /// // This artifact is outdated.
-/// let outdated_artifact = ArtifactId {
-///     version: "1.2.0".parse().unwrap(),
-///     ..valid_artifact.clone()
-/// };
+/// let outdated_artifact = ArtifactId::new(
```

Comment:
Nit: Could make sense to clone the `outdated_artifact` from `valid_artifact` and change its version.